pandas/core/reshape/reshape.py

```diff
@@ -689,7 +689,7 @@ def _convert_level_number(level_num, columns):
         new_labels = [np.arange(N).repeat(levsize)]
         new_names = [this.index.name]  # something better?
 
-    new_levels.append(level_vals)
+    new_levels.append(frame.columns.levels[level_num])
```

Comment:
I didn't - I swear I didn't. I don't even know what that means. I'll change it back, but I promise it wasn't me.